### PR TITLE
Adds 2.15 ee image and build update

### PR DESCRIPTION
**Pull request by anweshadas**
Updates the Github Action for the new execution environement file

### CODE REVIEW

**Comment by Spredzy** [comment]
Few comments here:

  1. Hard pinning on core should be removed. The latest available version should be pulled.
  2. I don't think this GHA is appropriate for this action. This one is triggered on push, PR and on schedule. For now and until we can eventually linked it to a new tag available on core, this should be run on-demand, with either the version specified in a form or even better introspected from the job to create the proper tag number to push to the registry with

File: execution-environments/2.15-with-ansible8/execution-environment.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
```
Comment (felixfontein) [inline execution-environments/2.15-with-ansible8/execution-environment.yml:1]: The directory name doesn't look good. This collection does **not** contain Ansible, it only contains a few collections that are part of Ansible.

**Comment by felixfontein** [comment]
The EE should also be listed in the EE readme: https://github.com/ansible-community/images/blob/main/execution-environments/README.md#available-images

**Comment by felixfontein** [comment]
> I don't think this GHA is appropriate for this action. This one is triggered on push, PR and on schedule.

It is appropriate. The GHA action makes sure that the EEs can actually be built. This needs to be checked regularly (cron), when PRs are created (they could modify the EE), and it's a good idea to also check that after merging a PR.

> For now and until we can eventually linked it to a new tag available on core automatically, this should be run on-demand, with either the version specified in a form or even better introspected from the job to create the proper tag number to push to the registry with

EEs should be built a lot more often than on releases, simply to make sure that updates to the underlying base image (like security updates) are incorporated frequently.

One big question is whether there should be multiple active tags for one EE at the same time, or not ("latest" only). If there are multiple tags, rebuilding them frequently requires a lot of resources.

**Comment by gotmax23** [comment]
> merged commit 31c0265 into main

https://github.com/ansible-community/images/pull/61#discussion_r1296362165 was never addressed.

**Comment by Andersson007** [comment]
> > merged commit [31c0265](https://github.com/ansible-community/images/commit/31c0265b4ad59e3b00f498b5d1876fcae7ce61b9) into main
> 
> [#61 (comment)](https://github.com/ansible-community/images/pull/61#discussion_r1296362165) was never addressed.

@gotmax23 thanks! @felixfontein apologies!
@anweshadas could you please take a look?

**Comment by felixfontein** [comment]
This PR doesn't look ready for merging to me.

The names don't match (see my comment), only one of the two EEs was added to CI, and the tests are over-specific (they test things that are not specified in the EE config file, like the exact Fedora version; and other things like the ansible version aren't checked at a ll).

Also for 2.15-with-ansible8 the collection versions are pinned, and there is no indication of how / when these will be updated. If we pin versions, we need a procedure to keep them up to date. (I would argue we should always use the latest non-prerelease.)

**Comment by felixfontein** [comment]
Also according to https://forum.ansible.com/t/community-execution-environment-image-content-and-location/262/14 the names should be `community-ee-minimal` and `community-ee-base`, and should not include `2.15` (since that will be wrong from 2023-11-06 on anyway, as the images do not specify a specific ansible-core version).

**Comment by gotmax23** [comment]
It looks like the images were [already published](https://github.com/orgs/ansible-community/packages?tab=packages&q=community-ee) last month without this PR being finalized, so it seems this process was done backwards.

**Comment by AlanCoding** [comment]
I'm glad to see this go in because it upgrades to the builder v3 format, and recommending anything built with the old base images is a bad idea. I think I agree with the other comments about naming, as it's hard to follow what the version numbers even refer to.

**Comment by Andersson007** [comment]
> It looks like the images were [already published](https://github.com/orgs/ansible-community/packages?tab=packages&q=community-ee) last month without this PR being finalized, so it seems this process was done backwards.

@gotmax23 yes, you're right, once we agreed in the forum topic about the content, we published those images to proceed with the Getting started EE guidelines. It was a blocker.

**Comment by felixfontein** [comment]
It has been two weeks since this PR was merged, but the problems mentioned here (that should have prevented a merge) haven't been fixed. @anweshadas @Andersson007

**Comment by Andersson007** [comment]
@felixfontein thanks for reminding! We remember about this and once @anweshadas 's back, she'll take a look

**Comment by felixfontein** [comment]
I created a follow-up PR to fix some of the issues in #65.
